json/src/spec/builtin.rs

```diff
@@ -16,50 +16,75 @@
 
 //! Spec builtin deserialization.
 
+use std::collections::BTreeMap;
+
 use crate::uint::Uint;
 use serde::Deserialize;
 
-
 /// Linear pricing.
 #[derive(Debug, PartialEq, Deserialize, Clone)]
 #[serde(deny_unknown_fields)]
 pub struct Linear {
 	/// Base price.
-	pub base: usize,
+	pub base: u64,
 	/// Price for word.
-	pub word: usize,
+	pub word: u64,
 }
 
 /// Pricing for modular exponentiation.
 #[derive(Debug, PartialEq, Deserialize, Clone)]
 #[serde(deny_unknown_fields)]
 pub struct Modexp {
 	/// Price divisor.
-	pub divisor: usize,
+	pub divisor: u64,
 }
 
 /// Pricing for constant alt_bn128 operations (ECADD and ECMUL)
 #[derive(Debug, PartialEq, Deserialize, Clone)]
 #[serde(deny_unknown_fields)]
 pub struct AltBn128ConstOperations {
 	/// price
-	pub price: usize,
+	pub price: u64,
 	/// EIP 1108 transition price
-	pub eip1108_transition_price: usize,
+	// for backward compatibility
+	pub eip1108_transition_price: Option<u64>,
+}
+
+impl AltBn128ConstOperations {
+	fn clear_eip1108(&mut self) {
+		self.eip1108_transition_price = None;
+	}
 }
 
 /// Pricing for alt_bn128_pairing.
 #[derive(Debug, PartialEq, Deserialize, Clone)]
 #[serde(deny_unknown_fields)]
 pub struct AltBn128Pairing {
 	/// Base price.
-	pub base: usize,
+	pub base: u64,
 	/// Price per point pair.
-	pub pair: usize,
+	pub pair: u64,
 	/// EIP 1108 transition base price
-	pub eip1108_transition_base: usize,
+	// for backward compatibility
+	pub eip1108_transition_base: Option<u64>,
 	/// EIP 1108 transition price per point pair
-	pub eip1108_transition_pair: usize,
+	// for backward compatibility
+	pub eip1108_transition_pair: Option<u64>,
+}
+
+impl AltBn128Pairing {
+	fn clear_eip1108(&mut self) {
+		self.eip1108_transition_base = None;
+		self.eip1108_transition_pair = None;
```

Comment:
Not sure what is the purpose of this function. This crate should be used only for parsing

Reply:
Yeah, that's not clear sorry (it's mess).  To give some context, see https://github.com/paritytech/parity-ethereum/pull/11039#discussion_r331566965

Thus, this crate if converts `old format` to `new_format` to avoid doing it elsewhere, ideally we should do it while deserializing and not have to do it at runtime to match `object` or `map`.

An edge case is when a `single activation` has `eip1108_transitions` then it's converted into a `BTreeMap` with two entries and sets the `eip1108 params` to `None` to make sure that those are used anymore.